libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/NavigationSession.kt

```diff
@@ -71,10 +69,24 @@ internal class NavigationSession : RoutesObserver, TripSessionStateObserver {
             TripSessionState.STOPPED -> false
         }
     }
+}
 
-    internal enum class State {
-        IDLE,
-        FREE_DRIVE,
-        ACTIVE_GUIDANCE
-    }
+/**
+ * Contains the various states that can occur during a navigation.
+ */
+enum class NavigationSessionState {
```

Comment:
Should this be a sealed class and have the session ID as a value?

Reply:
Came here to say this too. if the session id is in the observable state, we could delete `EventsSessionIdObserver` and only observe the `NavigationSession`.

We could surface a `NavigationSession` which has a `State` and an `id`. The class responsible for figuring this out would be `NavigationSessionController` (right now it's the `NavigationSession`)